Cleaned up documentation on `session` (#1522)
sveltejs/sapper: site/content/docs/07-state-management.md

</script>
```

* `preloading` contains a read-only boolean value, indicating whether or not a navigation is pending

> * `page` contains information on the current address. See [preloading](docs#Arguments) for details. This store is read-only.

Reviewer: ```suggestion
* `page` contains read-only information about the current route. See [preloading](docs#Arguments) for details.
```

* `preloading` contains a read-only boolean value, indicating whether or not a navigation is pending
* `page` contains information on the current address. See [preloading](docs#Arguments) for details. This store is read-only.
* `session` can be used to pass data from the server related to the current request. It is a [writable store](https://svelte.dev/tutorial/writable-stores), meaning you can update it with new data. If, for example, you populate the session with the current user on the server, you can update the store when the user logs in. Your components will refresh to reflect the new state

### Seeding session data

On the server, you can populate `session` by passing an option to `sapper.middleware`:
`session` is `undefined` by default. To populate it with data, implement a function that returns session data given an HTTP request. Pass it as an option to `sapper.middleware` when the server is started.

As an example, here is how to populate the session with the current user:

```js
// src/server.js

> express() // or Polka, or a similar framework

Reviewer: I would probably leave this comment for now so that people don't get confused by this saying `express` when the template uses `polka`. or maybe change `express` to `polka` here to match the template

Author: Oh. I was certainly confused by it. I thought this comment was part of the template code and wasn't relevant to the documentation. I'll switch to polka to match the sample then.

	.listen(process.env.PORT);
```

`req` is an `http.ClientRequest` and `res` an `http.ServerResponse`.

> The session data must be serializable. This means it must not contain functions or custom classes, just built-in JavaScript data types. [devalue](https://github.com/Rich-Harris/devalue) is used for serialization.

Reviewer: Maybe just make the reference to `devalue` a footnote. I would think most users shouldn't have to care about what's used internally

```suggestion
The session data must be serializable<sup>1</sup>. This means it must not contain functions or custom classes, just built-in JavaScript data types.
```

Then at the bottom of the page add:

1. [devalue](https://github.com/Rich-Harris/devalue) is used for serialization

Author: I'd suggest just removing it entirely. I agree it's an implementation detail.

The `session` function may return a `Promise` (or, equivalently, be `async`).

> > If `session` returns a `Promise`, every server-rendered page route will wait for it to resolve. This can slow down your application significantly.

Reviewer: I prefer the old text here for a few reasons. It starts with "Note that" to show it's something to pay attention to, informed users that `async` functions were also affected by this, it bolded every to call attention to it, it didn't try to predict the impact to the user's application

It was a little verbose though and I think could be made slightly more concise:

```suggestion
> Note that if `session` returns a `Promise` (or is `async`), it will be re-awaited for on **every** server-rendered page route.
```

Author: Sure. I just found the bolding shouty. I'm honestly not sure what it is really warning of. I assumed it was talking about the potential performance impact. It would be helpful to spell out why the await is or can be a problem.
